packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -294,11 +304,17 @@ class AOTSnapshotter {
       '-o', appLib,
       assemblyO,
     ];
-    final RunResult linkResult = await _xcode.clang(linkArgs);
-    if (linkResult.exitCode != 0) {
+    RunResult result = await _xcode.clang(linkArgs);
+    if (result.exitCode != 0) {
       _logger.printError('Failed to link AOT snapshot. Linker terminated with exit code ${compileResult.exitCode}');
+    } else if (splitDebugInfo != null) {
+      final String dSYMPath = _fileSystem.path.join(splitDebugInfo, targetArch, 'App.dSYM');
+      result = await _xcode.dsymutil(<String>['-o', dSYMPath, appLib]);
```

Comment:
This needs unit tests.  Would look something like this
https://github.com/flutter/flutter/blob/4cf60f7659cba379200b7d80ceb45d020dae560e/packages/flutter_tools/test/general.shard/base/build_test.dart#L273

Reply:
I've updated those tests. I'll also have a look at `build_system/targets/ios_test.dart`